packages/core/src/types.ts

```diff
@@ -102,8 +103,14 @@ export type SettingsConfig = Record<string, unknown>;
 /**
  * The configuration for a rule.
  */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- This is a generic type
```

Comment:
> generic

_"generic"_ has an existing meaning in TypeScript code: it's about constructs with type parameters, like `type Something<T> = ...`. I'm guessing this meant that the type is _"general"_(ized)?

This type being general isn't a reason to use `any` though. See comment on next line.

I'd recommend looking through the `typescript-eslint` types -e.g. https://github.com/typescript-eslint/typescript-eslint/blob/510153b5c4fa0eef5b9e1e7e87c29162a2601228/packages/utils/src/ts-eslint/Config.ts- to see how we do things. A lot of these types are repeat ones.